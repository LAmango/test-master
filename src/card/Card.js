import React from "react";
import styled from "styled-components";

const Content = styled.div`
  color: black;
  width: 500px;
  padding: 140px 0;
  text-align: center;
  height: 100%;
`;

const Counter = styled.div`
  position: absolute;
  right: 0;
  padding: 10px;
  color: gray;
`;

class Card extends React.Component {
  render() {
    const { cardSet, side, current, total } = this.props;
    return (
      <div>
        {cardSet && (
          <Counter>
            {current}/{total}
          </Counter>
        )}
        <Content>{side}</Content>
      </div>
    );
  }
}

export default Card;
