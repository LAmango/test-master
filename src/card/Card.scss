.paper {
  margin: auto;
  width: 100%;
  height: 300px;
  cursor: pointer;
}

.flip {
  will-change: transform, opacity;
  position: absolute;
}
