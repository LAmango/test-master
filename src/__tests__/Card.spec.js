import React from "react";
import { create } from "react-test-renderer";
import Card from "../card/Card";

describe("Card Component", () => {
  it("Matches the snapshot", () => {
    const card = create(<Card />);
    expect(card.toJSON()).toMatchSnapshot();
  });

  it("Doesn't show current or total cards if a card set hasn't been chosen", () => {
    const component = create(<Card cardSet={null} />);
    const card = component.root;
    const childDiv = card.findByType("div");
    expect(childDiv.children.length).toEqual(1);
    expect(typeof card.props.side).toEqual("undefined");
  });

  it("Show current or total cards if a card set hasn't been chosen", () => {
    const component = create(<Card cardSet={"colors"} />);
    const card = component.root;
    const childDiv = card.findByType("div");
    expect(childDiv.children.length).toEqual(2);
  });

  it("Show current or total cards if a card set has been chosen", () => {
    const component = create(<Card cardSet={"colors"} />);
    const card = component.root;
    const childDiv = card.findByType("div");
    expect(childDiv.children.length).toEqual(2);
  });
});
