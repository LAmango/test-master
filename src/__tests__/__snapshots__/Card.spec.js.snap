// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Matches the snapshot 1`] = `
<div>
  <div
    className="sc-bdVaJa usiNU"
  />
</div>
`;
